### PR TITLE
Solver unit test

### LINKED ISSUE #66: Solver API unit tests
##### Description
write unit tests for the [solver](https://github.com/operator-framework/deppy/tree/main/internal/solver) package

##### A/C
- unit tests written with good coverage (happy path, edge cases, etc.
- being executed by CI (?)

### PR DESCRIPTION
**Pull request by akihikokuroda**
This PR has some solver unit tests.  Are they good form for the solver unit test?  If so I can add some more tests.  

The unit tests are in : pkg/solver/solver_test.go
Closes #66

### CODE REVIEW

File: pkg/solver/solver_test.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	entitysource.EntityQuerier
+	entitysource.EntityContentGetter
+}
+
+func (c EntitySourceStruct) GetVariables(ctx context.Context, querier entitysource.EntityQuerier) ([]sat.Variable, error) {
+	return c.variables, nil
+}
+
+func createEntities(num int) map[entitysource.EntityID]entitysource.Entity {
+	entities := make(map[entitysource.EntityID]entitysource.Entity)
```
Comment (perdasilva) [inline pkg/solver/solver_test.go:30]: maybe pass `i` in as the capacity: `make(map[entitysource.EntityID]entitysource.Entity, i)`

File: pkg/solver/solver_test.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	"github.com/operator-framework/deppy/pkg/entitysource"
+	entitysourcefactory "github.com/operator-framework/deppy/pkg/entitysource/factory"
+	"github.com/operator-framework/deppy/pkg/sat"
+	satconstraints "github.com/operator-framework/deppy/pkg/sat/constraints"
+	"github.com/operator-framework/deppy/pkg/solver/factory"
+)
+
+type EntitySourceStruct struct {
+	variables []sat.Variable
+	entitysource.EntityQuerier
```
Comment (perdasilva) [inline pkg/solver/solver_test.go:21]: maybe just use `entitysource.EntitySource`?

File: pkg/solver/solver_test.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+func (c EntitySourceStruct) GetVariables(ctx context.Context, querier entitysource.EntityQuerier) ([]sat.Variable, error) {
+	return c.variables, nil
+}
+
+func createEntities(num int) map[entitysource.EntityID]entitysource.Entity {
+	entities := make(map[entitysource.EntityID]entitysource.Entity)
+	for i := 0; i < num; i++ {
+		id := entitysource.EntityID(string([]byte{byte('1' + i)}))
```
Comment (perdasilva) [inline pkg/solver/solver_test.go:32]: suggestion: `id := entitysource.EntityId(strconv.Itoa(i + 1))`

Comment (akihikokuroda) [reply pkg/solver/solver_test.go:32]: I'll update.  Thanks!

File: pkg/solver/solver_test.go (hunk 1 above)
Comment (akihikokuroda) [reply pkg/solver/solver_test.go:30]: I'll update.  Thanks!

**Comment by akihikokuroda** [comment]
@perdasilva Thanks for review.  I updated the tests and rebased for the current `main` branch.

**Review by perdasilva — APPROVED** [review]
lgtm
